src/number/complete.rs

```diff
@@ -1426,11 +1427,13 @@ where
   )(input)
 }
 
+///
+
 /// Recognizes a floating point number in text format and returns the integer, fraction and exponent parts of the input data
 ///
 /// *Complete version*: Can parse until the end of input.
 ///
-pub fn recognize_float_parts<T, E: ParseError<T>>(input: T) -> IResult<T, (bool, T, T, i32), E>
+pub fn recognize_float_parts<T, E: ParseError<T>>(input: T) -> IResult<T, (T, T, i32), E>
```

Comment:
That breaking change unfortunately, if we want to avoid the double check of the sign we will need to factorize this function to not make breaking change.

Reply:
Is recognize_float_parts part of the public API? If so, I can change that...

Reply:
apparently, never used it though... I don't even understand why it's pub, https://docs.rs/nom/7.0.0/nom/number/complete/fn.recognize_float_parts.html

Reply:
Ok, I guess it is. Then there's no real way to avoid using `sign!()` twice, since we don't return the sign if recognized the float parts errors, so we just have to trust the compiler inlines it.

Reply:
I wonder if it was just an implementation detail that was exposed publicly, and now can't be reverted?